website/src/controllers/api-v2/groups.js

```diff
@@ -445,12 +452,14 @@ api.likeChatMessage = function(req, res, next) {
   } else {
     message.likes[user._id] = true;
   }
+  group.chat = chat;
   group.markModified('chat');
   group.save(function(err,_saved){
     if (err) return next(err);
     // @TODO: We're sending back the entire array of chats back
     // Should we just send back the object of the single chat message?
     // If not, should we update the group chat when a chat is liked?
+    _purgeFlagInfoFromChat(_saved, user);
```

Comment:
Since we send back the whole chat array, we have to re-purge it of flagged messages :(